lib/chef/knife/vsphere_vm_snapshot.rb

```diff
@@ -94,7 +94,7 @@ def run
     if get_config(:list) && vm.snapshot
       puts 'Current snapshot tree: '
       puts "#{vmname}"
-      snapshot_list.each { |i| puts display_node(i, current_snapshot) }
+      snapshot_list.each { |i| ui.output(display_node(i, current_snapshot)) }
```

Comment:
I'm thinking about how the JSON will look here... A couple of thoughts.

First is those two `puts` may cause problems with parsing because they're not JSON and will show up in any format. For example:

```
sean~/play/knife-vsphere (nammiesgal-master)$ be knife vsphere vm snapshot --list rolebasetest -F json  | jq .
parse error: Invalid numeric literal at line 1, column 8
```

So I think we need to get rid of them so our output has a valid format no matter what's chosen. One option would be to ditch them entirely, the other would be to make the output a Hash with the key of the vm name and the value of the snapshots. This also dovetails into my second concern.

If `snapshot_list` has multiple elements (I wasn't able to get it to do so here, but since it's an array it must be a possibility), then the resulting JSON will be a series of hashes not enclosed by an array, e.g.

```
{ snap 1 }
{ snap 2 }
```

when we probably need something like:

```
[
  { snap1 },
  { snap2 },
]
```

You could get this fairly easily by using a `#map` instead of `#each`, e.g.

`ui.output snapshot.map { |snap| display_node(snap, current_snapshot) }`

From there if you decided you wanted the top level to be a key you could have:

`ui.output { vmname: snapshot.map { |snap| display_node(snap, current_snapshot) } }`

Reply:
ok, I implemented this as an array of snapshots without the 2 put statements.  Looks good: 
![image](https://user-images.githubusercontent.com/23378716/37552353-0e44500e-29ff-11e8-802e-272f4ab1e594.png)

![image](https://user-images.githubusercontent.com/23378716/37552355-2098616e-29ff-11e8-993e-f0ce222c0dda.png)

Reply:
Note that in the 2nd image, the snapshot named Tree_Test is duplicated because I mistakenly screenshotted it again when I should have left it out!